### ISSUE TITLE
build “4.3.0+2514~gcd43e9e070” for Windows: lens correction on lensfun not OK 

### ISSUE THREAD

**Issue by cabuy** [darktable-org/darktable#14696]
### Describe the bug

I am testing dt nightly build “4.3.0+2514~gcd43e9e070” for Windows (upgrading from dt 4.2.1).
The lens correction module does not find the right lens from the Lensfun database. 
So far (till dt 4.2.1) the lens corrections on all my pictures are from lensfun. 
But in this build lensfun seems to retrieve the wrong camera and the wrong lens.
Examples
MAC60615 taken with camera Fujifilm X-E1 and lens XF27mmF2.8, lens correction on lensfun gives camera X-E2 and lens XF18-55mm.
MAC32550 taken with camera Fujifim X-T2 and lens XF10-24mmF4, lens correction on lensfun gives camera X-E2 and lens XF18-55mm.
Can you please investigate,
Marc.

PS good news is that the lens corrections can be handled for these fujifilm camera's through the metadata now  (Issue #13358)

### Steps to reproduce

I attach the 2 raw files and XMP files. The exif info gives the correct camera en lens, but lens-correction-lensfun is not OK

### Expected behavior

lens correction should retrieve the correct lens as before

### Logfile | Screenshot | Screencast

[lens correction on lensfun.zip](https://github.com/darktable-org/darktable/files/11714841/lens.correction.on.lensfun.zip)
[lens correction on lensfun1.zip](https://github.com/darktable-org/darktable/files/11714842/lens.correction.on.lensfun1.zip)

### Where did you install darktable from?

distro packaging

### darktable version

4.3.0+2514~gcd43e9e070

### What OS are you using?

Windows

### What is the version of your OS?

Windows 10 pro

### Describe your system?

I just upgraded from stable 4.2.1 and encountered this issue - so I assume the system info is not that relevant

### Are you using OpenCL GPU in darktable?

Yes

### If yes, what is the GPU card and driver?

NVIDIA Geforce RTX 2080 with Max-Q  -  NVIDIA driver 535.98

**Comment by gi-man** [comment]
> ### Where did you install darktable from?
> distro packaging

There is no distro packing for windows. Where are you getting darktable install files? Self build?

**Comment by wpferguson** [comment]
Looks like this is the nightly build artifact

**Comment by cabuy** [comment]
When discussing issue #13358 on github, I was pointed to the nightly Windows build at https://github.com/darktable-org/darktable/releases/tag/nightly. 
I downloaded the latest nightly Windows build (4.3.0+2514~gcd43e9e070) today and after upgrading from 4.2.1, I 
saw most of my previews in the lighttable going wrong because of the lens correction module not retrieving camera/lens profiles.
Sorry for the misleading “distro” name.
Regards, Marc.

PS There was also talk about the weekly Windows build, somewhere on discuss.pixls.us but I haven’t located that one.

**Comment by wpferguson** [comment]
Perhaps it would be better to test with the weekly builds from pixls.us (https://discuss.pixls.us/t/darktable-windows-insider-program-6-4/37407) which are built with the release packaging.

A new build will be out in a few hours.

**Comment by cabuy** [comment]
OK, I'll try that later today and let you know how it's going.

**Comment by cabuy** [comment]
Couldn't wait. This weekly build version "4.3.0+2461~g2f2c69e9cc" for Windows shows the SAME PROBLEM. You can probably simulate by using the raw and xmp in the 2 zip files attached yo this issue notification.
I will be happy to further test with new builds. Just let me know.

**Comment by gi-man** [comment]
Opening the image with the associated xmp, it opens with the wrong lens. Fedora 37 4.3.0~git2508.b320d1ed

When I reset the module, it switches to embedded metadata, which looks like a correct lens profile. When I then select Lensfun database, it selects the correct lens (10-24...)

**Comment by wpferguson** [comment]
If I import the images without the xmp files then lens correction goes to embedded metadata.  If I change to lensfun database, then both images show the correct camera and lens (per the above).  This was on Linux (Ubuntu 22.04).

**Comment by wpferguson** [comment]
Tried on Windows 10 with weekly build from last Sunday.  Imported without the xmp files.  When I opened the image, then lens correction, it showed the embedded metadata.  When I selected lensfun, it showed the correct camera and lens.

Do you have any presets for lens correction?

**Comment by cabuy** [comment]
No, I have no presets.

**Comment by cabuy** [comment]
I only have a preset for demosaicing Fujifilm.

**Comment by cabuy** [comment]
I have multiple images with this issue. For some of them I can already see the wrong lens correction in the Lighttable thumbnails.
Now it acts for me in the same manner as for gi-man. So
1. picture shows up with wrong camera/lens from lensfun
2. reset lens correction > metadata are chosen
3. choose lensfun > correct camera/lens is found.

**Comment by gi-man** [comment]
Do you still have 4.2.1 installed? Can you confirm the correct lens is part of the xmp in 4.2.1?

**Comment by cabuy** [comment]
I have never seen this problem before so yes the correct lens is part of the xmp in 4.2.1. Also what is strange: he always shows up with this "camera X-E2 and lens XF18-55mm" combination.

**Comment by cabuy** [comment]
The other observation : a new import comes straight with lens correction on metadata all. 
So I guess this may have to do with the modifications made to the lens correction recently, like the one for issue #13358. 
In 4.2.1 the only camera model that I had supporting the metadata "all" is the X-T4. All my X-T4 pictures were imported with lensfun lens corrections and they still show the right lens. Now metadata "all" can be chosen  for different Fujifilm camera's. For all those new ones the lens/camera combination comes up with the wrong "camera X-E2 and lens XF18-55mm" combination.

**Comment by wpferguson** [comment]
>   For all those new ones the lens/camera combination comes up with the wrong "camera X-E2 and lens XF18-55mm" combination.

When I imported, on Linux and Windows, the imports came with metadata all.  When I changed to lensfun, the correct camera and lens were present.  I never saw "camera X-E2 and lens XF18-55mm".

**Comment by wpferguson** [comment]
The included xmp files are from dt 4.2.1?

**Comment by wpferguson** [comment]
@TurboGit, @sgotti 

Two zips are included above which contain fuji files edited in dt 4.2.1 with lens correction applied using the lensfun database.  When those files are imported into current master they show the wrong camera model and lens.

Commit f51dee797bef0fcde490c70312363c80e993bf65 breaks the first file.  Commit f843de8f24d935874f0697a582acc9e87c6c8653 breaks the second file.

If you reset the module, then the module picks up the embedded metadata.  If you then change to the lensfun database, the camera and lens are correctly identified.

To sum up this bug appears to break dt 4.2.1 edits for fuji pics where lens correction was applied.

**Comment by TurboGit** [comment]
@sgotti : The source cut for 4.4 is planed on Wed 14th, do you think you can fix that before? Let me know, thanks.

**Comment by cabuy** [comment]
Sorry Pascal for not understanding the internals from dt. If this means that my xmp's are corrupted than that may be the case for about 80% of my edits. I've been using dt since version 3.02 and have always used the official releases except now.  I can only confirm that this problem has never appeared before.

**Comment by TurboGit** [comment]
>  If this means that my xmp's are corrupted

Not corrupted as the XMP does contains a valid camera/lens but I don't understand why this does not match the actual RAW camera/lens! At least to me the pointed wrong commit are not the culprit. Let's see if @sgotti has a better idea about what could have caused this.

**Comment by cabuy** [comment]
Let me know if/how I can help. For instance I can provide the 4.2.1 xmp's before they we're opened under this 4.3 version.

**Comment by TurboGit** [comment]
@sgotti : To me this issue comes from the handling of old `modified` & `modify_flags` in params.

For MAC32550 with 4.2 on the history tooltip I see that we have the changes:
```
camera "X-E2"                     -> "X-T2"
len    "XF18-55mmF2.8-4 R LM OIS" -> "XF10-24mmF4 R OIS"
modify 0                          -> 1
```

So we have triggered the update from lensfun database (modify above) and this doesn't happen on current master. That's probably the issue, and this is release critical.

**Comment by sgotti** [comment]
@TurboGit in these days I don't have much spare time. But I don't understand how this is related to the changes on the embedded metadata algo. Isn't this related to the changes introduced by 5e75b113013fd98b625b1d0fd624f7331e13742b ?

**Comment by TurboGit** [comment]
@wpferguson @cabuy : See #14705. I'll appreciate a review and/or testing if possible. We are very close from the release so we need to be very cautious. TIA.

**Comment by TurboGit** [comment]
@sgotti : Thanks for hinting in the right direction.

**Comment by cabuy** [comment]
wpferguson: I'll be happy to retest as soon as somebody warns me that it was included in the nightly build for Windows. Or of you like I can provide the genuine 4.2.1 xmp's of both images attached here earlier.

**Comment by cabuy** [comment]
wpferguson: I'll be happy to retest as soon as somebody warns me that it was included in the nightly build for Windows. Or of you like I can provide the genuine 4.2.1 xmp's of both images attached here earlier.

**Comment by wpferguson** [comment]
I took the original images and opened them in 4.2.1.  I applied lens correction and got the correct values.  I then imported those files and xmps into current master and they loaded correctly.  This is on Unbuntu.  Tested on windows with last week's weekly build and the 4.2.1 xmp files imported correctly.  

So, it appears that the attached xmp files are corrupt in some manner.  @TurboGit I've attached the image files with the xmp files generated by dt 4.2.1 if you want to test too.
[lensfun.4.2.1-2.zip](https://github.com/darktable-org/darktable/files/11725862/lensfun.4.2.1-2.zip)
[lensfun.4.2.1-1.zip](https://github.com/darktable-org/darktable/files/11725864/lensfun.4.2.1-1.zip)

**Comment by TurboGit** [comment]
> So, it appears that the attached xmp files are corrupt in some manner.

I don't think so, this is a tricky issue that took me time to understand. To me the issue arise only when an image is not edited and a copy/paste of a lens correction module from one image with a different camera/lens. In this case there was a circuitry to record that the camera/lens are in fact not initialized (even if some values are there, wrong values as reported here). In this case the values are reloaded (reset) using a check on the lensfun database with proper camera/lens. That was the only option before the be able to copy/paste lens correction module on a set of images with different camera/lens. A hack that I thought was OBE and so I had removed the code to handle this corner case.

**Comment by wpferguson** [comment]
I tested 14705 and the problem still exists.

**Comment by wpferguson** [comment]
14705 works for me

**Comment by cabuy** [comment]
Ok, then let me know from where to download it.

**Comment by wpferguson** [comment]
@cabuy https://drive.google.com/file/d/16M9Ez2bE3TluirWlMZVAIKj27LE0CrP4/view?usp=sharing

**Comment by TurboGit** [comment]
@wpferguson : Thanks for your precious help on this issue.

**Comment by wpferguson** [comment]
@TurboGit, you're welcome.

@sgotti sorry for the bad bisect.  Modified my bisect procedure to make sure I have a clean environment for each run

**Comment by cabuy** [comment]
@sgotti @wpferguson @Turbogit : IT WORKS !!!

As an enthusiast darktable user I want to thank you all for implementing the lens corrections through metadata for the Fujifilm X-Trans camera’s (sgotti Issue #13358) and getting this current issue fixed. 
I must admit that the thought of NOT using an official release on my only darktable environment scared me. But with this prompt support to get this solved, I hope do it again in the last month of dt 4.5. 

The option (lens corrections through metadata) isn’t necessarily for my immediate personal interest. But, as Fujifilm opened up their lens protocols to cheaper 3rd parties (Sigma, Tamron, …) it makes the use of darktable more attractive for Fujifilm X-Trans shooters. They don’t need to consult the lensfun support list anymore.

Thank you and kind regards,
Marc (Belgium)
